### ISSUE TITLE
Fix incorrect or incomplete Javadoc in public APIs

### ISSUE THREAD

**Issue by jxblum** [spring-projects/spring-data-commons#2602]
Many of the classes (and APIs) in the Spring Data Commons core codebase have incomplete or incorrect Javadoc.

**Comment by aml8801** [comment]
Is this  Issue still open? Could you give me hint about the standard, which should be used?

**Comment by jxblum** [comment]
Fixing _Javadoc_ in _Spring Data Commons_ will most likely be a multi-part development effort.

A good first step would be to simply make sure that any _Javadoc_ on a class or method is complete, i.e. does not have empty _Javadoc_ tags for type parameter(s)/variable(s), method parameters (`@param`), `@return` values, `@throws` clauses, etc.  A informative and proper description should be added wherever there are missing Javadoc, which the IDE flags as a warning.

Here is 1 such [example](https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/repository/RepositoryDefinition.java#L45).

This ([here](https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/repository/CrudRepository.java#L34)) is incomplete. This is, it does not "describe" the `entity`.

Here is another [example](https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/domain/Range.java#L68-L71).

These type of _Javadoc_ problems are littered throughout the codebase.

**Comment by ukasus** [comment]
Hi , is this issue still open?

**Comment by jxblum** [comment]
Yes, this is an ongoing effort. As a team, we decided to make changes to the Javadoc as we go, addressing issues or enhancements in the codebase along the way. Therefore, I would expect it would take sometime before all the Javadoc is completely reviewed and edited since it is not a focused effort.
